Standalone/Standalone/NativeTableViewController.m

```diff
@@ -0,0 +1,145 @@
+    [self.tableView registerNib:[UINib nibWithNibName:@"NativeAdTableViewCell" bundle:nil] forCellReuseIdentifier:@"nativeAdCell"];
+    
+    [self addContent];
+    [self addNativeAd];
+}
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.tableViewItems.count;
+}
+
+- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
+    if ([self.tableViewItems[indexPath.row] isKindOfClass:[NativeAdTableViewCell class]]) {
+        return self.nativeAdLoadStatus ? cellHeight : 0;
+    }
+    return cellHeight;
+}
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    if ([self.tableViewItems[indexPath.row] isKindOfClass:[NativeAdTableViewCell class]]) {
+        UITableViewCell *cell = [self.tableView dequeueReusableCellWithIdentifier:@"nativeAdCell" forIndexPath:indexPath];
+        for (UIView *subview in cell.contentView.subviews) {
+            [subview removeFromSuperview];
+        }
+        NativeAdTableViewCell *nativeAdCell = self.tableViewItems[indexPath.row];
+        [cell.contentView addSubview:nativeAdCell];
+        return cell;
+    } else {
+        ContentItem *item = self.tableViewItems[indexPath.row];
+        ContentItemTableViewCell *contentItemCell = [self.tableView dequeueReusableCellWithIdentifier:@"contentCell" forIndexPath:indexPath];
+        contentItemCell.titleLabel.text = item.contentTitle;
+        contentItemCell.descriptionLabel.text = item.contentDescription;
+        contentItemCell.contentImage.image = item.contentImage;
+        return contentItemCell;
+    }
+}
+
+- (void)addContent {
+    UIImage *image = [UIImage imageWithData:[NSData dataWithContentsOfURL:[NSURL URLWithString:@"https://img.icons8.com/cotton/2x/news.png"]]];
+    for (int i = 0; i < 10; i++) {
+        ContentItem *item = [[ContentItem alloc] init];
+        item.contentTitle = [NSString stringWithFormat:@"%d. Lorem ipsum dolor sit amet", i + 1];
+        item.contentDescription = @"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed elementum augue dignissim mi ullamcorper cursus. Nulla varius vulputate elit. Nullam..";
+        item.contentImage = image;
+        [self.tableViewItems addObject:item];
+    }
+}
+
+- (void)addNativeAd {
+    self.nativeLoader = [[CRNativeLoader alloc] initWithAdUnit:[AdConfigurations criteoNativeAdUnit]];
+    self.nativeLoader.delegate = self;
+    [self.nativeLoader loadAd];
+    
+    [self.tableView layoutIfNeeded];
+    self.nativeAdCell = [[NSBundle mainBundle] loadNibNamed:@"NativeAdTableViewCell" owner:nil options:nil].firstObject;
+    [self.tableViewItems insertObject:self.nativeAdCell atIndex:nativeAdIndex];
+    self.nativeAdLoadStatus = NO;
```

Comment:
Is it better to insert the native object on `nativeLoader:didReceiveAd:` so that you don't have to maintain `nativeAdLoadStatus`?

Reply:
Thanks. I moved the code to `nativeLoader:didReceiveAd:` and also delete some inside `heightForRowAtIndexPath`.